### PR TITLE
Numpyro

### PR DESCRIPTION
**Pull request by thomaspinder**
This PR ports the existing Distrax functionality into NumPyro according to #121 

## Pull request type

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ x ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

Issue Number: #121

### CODE REVIEW

File: gpjax/gps.py [hunk 1]
```diff
@@ -198,16 +195,16 @@ class ConjugatePosterior(AbstractPosterior):
     jitter: Optional[float] = DEFAULT_JITTER
 
     def predict(
-        self, train_data: Dataset, params: Dict
-    ) -> Callable[[Float[Array, "N D"]], dx.Distribution]:
+        self, train_data: Dataset, params: dict
+    ) -> Callable[[Float[Array, "N D"]], npd.Distribution]:
```
Comment (Daniel-Dodd) [inline gpjax/gps.py:199]: Dictionary typing needs to be consistent - your changes have replaced some methods with `Dict` to `dict`, while others are unchanged . I opt to us `Dict` as it more general, though happy to switch to `dict` if there's good reason - though all methods need to be changed throughout the library.

File: examples/collapsed_vi.pct.py [hunk 2]
```diff
@@ -86,33 +86,32 @@
...
+parameter_state = gpx.initialise(sgpr, key)
+params, trainables, bijectors = parameter_state.unpack()
 
-loss_fn = jit(sgpr.elbo(D, constrainers, negative=True))
+loss_fn = jit(sgpr.elbo(D, negative=True))
 
 optimiser = ox.adam(learning_rate=0.005)
 
-params = gpx.transform(params, unconstrainers)
+params = gpx.unconstrain(params, bijectors)
```
Comment (Daniel-Dodd) [inline examples/collapsed_vi.pct.py:96]: We don't need an `unconstrain` here.

File: examples/collapsed_vi.pct.py [hunk 3]
```diff
@@ -166,15 +165,15 @@
 
 # %%
 full_rank_model = gpx.Prior(kernel=gpx.RBF()) * gpx.Gaussian(num_datapoints=D.n)
-fr_params, fr_trainables, fr_constrainers, fr_unconstrainers = gpx.initialise(
+fr_params, fr_trainables, fr_bijectors = gpx.initialise(
     full_rank_model, key
 ).unpack()
-fr_params = gpx.transform(fr_params, fr_unconstrainers)
-mll = jit(full_rank_model.marginal_log_likelihood(D, fr_constrainers, negative=True))
+fr_params = gpx.unconstrain(fr_params, fr_bijectors)
```
Comment (Daniel-Dodd) [inline examples/collapsed_vi.pct.py:171]: Similar comment about `unconstrain`.

File: examples/classification.pct.py [hunk 4]
```diff
@@ -72,37 +70,34 @@
 # To begin we obtain a set of initial parameter values through the `initialise` callable, and transform these to the unconstrained space via `transform` (see the [regression notebook](https://gpjax.readthedocs.io/en/latest/nbs/regression.html)). We also define the negative marginal log-likelihood, and JIT compile this to accelerate training.
 # %%
 parameter_state = gpx.initialise(posterior)
-params, trainable, constrainer, unconstrainer = parameter_state.unpack()
-params = gpx.transform(params, unconstrainer)
+params, trainable, bijectors = parameter_state.unpack()
+params = gpx.unconstrain(params, bijectors)
```
Comment (Daniel-Dodd) [inline examples/classification.pct.py:75]: Similar comment about `unconstrain`.

File: examples/classification.pct.py [hunk 5]
```diff
@@ -150,7 +145,7 @@
 L_inv = jsp.linalg.solve_triangular(L, I(D.n), lower=True)
 H_inv = jsp.linalg.solve_triangular(L.T, L_inv, lower=False)
 
-laplace_approximation = dx.MultivariateNormalFullCovariance(f_map_estimate, H_inv)
+laplace_approximation = numpyro.distributions.MultivariateNormal(f_map_estimate, H_inv)
```
Comment (Daniel-Dodd) [inline examples/classification.pct.py:148]: Would be nice to import `numpyro.distributions` as `npd` throughout this notebook.

**Review by Daniel-Dodd — COMMENTED** [review]
Great PR. Main comment is about `Dict` vs `dict` consistency. Issues with docs have already been address at my end, so ignore these comments - they are mainly for myself when I come to rebase.

File: gpjax/gps.py (hunk 1 above)
Comment (thomaspinder) [reply gpjax/gps.py:199]: These have been reolved now.

**Pull request merged**